rocker-org/rocker-versioned2
# Add a experimental script for batch creation of multiple user accounts in RStudio Server

#### Reference Issues
The proposed approach partially addresses the multi-user feature request at #160.

#### What does this implement/fix?
This pull request adds basic multi-user support to a RStudio server via a batch creation process. The credentials of multiple users, which should be stored in an environment variable (i.e., `BATCH_USER_CREATION=user1:pass1;user2:pass2;...`), are processed when the container starts. This simple feature can be used, for example, to rapidly setup a RStudio sandbox server for educational purposes (I'm using it during laboratory activities in the Database Systems course for students of the Statistics major at the Federal University of Espírito Santo ([UFES](https://ufes.br/), Brazil).

#### How to test it?
1. Open a Terminal window (e.g., Bash) and build a docker image using the following command:

        docker build . -f tests/rstudio/Dockerfile_rstudio_batch-users -t rocker/rstudio:batch-users

1. Then, start the container:

        docker run -d -p 8787:8787 -e PASSWORD=yourpasswordhere -e BATCH_USER_CREATION="user1:pass1;user2:pass2;user3:pass3" rocker/rstudio:batch-users

    or alternatively,

        docker run -d -p 8787:8787 -e PASSWORD=yourpasswordhere --env-file tests/rstudio/env.batch-users rocker/rstudio:batch-users

1. Finally, open a Web browser window and visit the URL http://localhost:8787. From there, the given users should be able to use the same RStudio server instance independently.

#### Other Comments
Feel free to suggest other approaches to address the problem/feature. Next, I plan to investigate the integration with LDAP user authentication.
